src/core/agent/services/multiSigService.ts

```diff
@@ -770,6 +777,149 @@ class MultiSigService extends AgentService {
     }
     return true;
   }
+
+  async getMultisigParticipants(multisigSignifyName: string) {
+    const members = await this.props.signifyClient
+      .identifiers()
+      .members(multisigSignifyName);
+    const multisigMembers = members["signing"];
+    let ourIdentifier;
+    for (const member of multisigMembers) {
+      const identifiers =
+        await this.identifierStorage.getMultisigMembersMetadata(member.aid);
+      if (identifiers.length) {
+        ourIdentifier = identifiers[0];
+        break;
+      }
+    }
+    if (!ourIdentifier) {
+      throw new Error(MultiSigService.MEMBER_AID_NOT_FOUND);
+    }
+    return {
+      ourIdentifier,
+      multisigMembers,
+    };
+  }
+
+  async endRoleAuthorization(multisigSignifyName: string) {
+    const { ourIdentifier, multisigMembers } =
+      await this.getMultisigParticipants(multisigSignifyName);
+    const hab = await this.props.signifyClient
+      .identifiers()
+      .get(multisigSignifyName);
+    const aid = hab["prefix"];
+    const results = [];
+    for (const member of multisigMembers) {
+      const eid = Object.keys(member.ends.agent)[0]; //agent of member
+      const stamp = new Date().toISOString().replace("Z", "000+00:00");
+
+      const endRoleRes = await this.props.signifyClient
+        .identifiers()
+        .addEndRole(multisigSignifyName, "agent", eid, stamp);
+      const op = await endRoleRes.op();
+      const rpy = endRoleRes.serder;
+      const sigs = endRoleRes.sigs;
+      const mstate = hab["state"];
+      const seal = [
+        "SealEvent",
+        { i: hab["prefix"], s: mstate["ee"]["s"], d: mstate["ee"]["d"] },
+      ];
+      const sigers = sigs.map((sig) => new Siger({ qb64: sig }));
+      const roleims = d(
+        messagize(rpy, sigers, seal, undefined, undefined, false)
+      );
+      const atc = roleims.substring(rpy.size);
+      const roleEmbeds = {
+        rpy: [rpy, atc],
+      };
+
+      const recp = multisigMembers
+        .filter((signing: any) => signing.aid !== ourIdentifier.id)
+        .map((member: any) => member.aid);
+      const ourAid: Aid = await this.props.signifyClient
+        .identifiers()
+        .get(ourIdentifier.signifyName as string);
```

Comment:
The calculations for `recp` and `ourAid` can be done outside the while loop commonly I believe